centreon/src/Core/AgentConfiguration/Application/UseCase/FindAgentConfiguration/FindAgentConfiguration.php

```diff
@@ -0,0 +1,124 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\AgentConfiguration\Application\UseCase\FindAgentConfiguration;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\AgentConfiguration\Application\Exception\AgentConfigurationException;
+use Core\AgentConfiguration\Application\Repository\ReadAgentConfigurationRepositoryInterface;
+use Core\AgentConfiguration\Domain\Model\Poller;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\ResponseStatusInterface;
+use Core\MonitoringServer\Application\Repository\ReadMonitoringServerRepositoryInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+
+final class FindAgentConfiguration
+{
+    use LoggerTrait;
+
+    /**
+     * FindAgentConfiguration constructor.
+     *
+     * @param ContactInterface $user user requesting the agent configuration
+     * @param ReadAgentConfigurationRepositoryInterface $readRepository repository to read agent configurations
+     * @param ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository repository to read monitoring servers
+     * @param ReadAccessGroupRepositoryInterface $readAccessGroupRepository repository to read access groups
+     */
+    public function __construct(
+        private readonly ContactInterface $user,
+        private readonly ReadAgentConfigurationRepositoryInterface $readRepository,
+        private readonly ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+    ) {
+    }
+
+    /**
+     * Retrieves an agent configuration with associated pollers.
+     *
+     * @param int $agentConfigurationId
+     *
+     * @return FindAgentConfigurationResponse|ResponseStatusInterface
+     */
+    public function __invoke(int $agentConfigurationId): FindAgentConfigurationResponse|ResponseStatusInterface {
+        $this->info(
+            'Find agent configuration',
+            [
+                'user_id' => $this->user->getId(),
+                'agent_configuration_id' => $agentConfigurationId,
+            ]
+        );
+
+        try {
+            if (null === $agentConfiguration = $this->readRepository->find($agentConfigurationId)) {
+                $this->error(
+                    'Agent configuration not found',
+                    ['agent_configuration_id' => $agentConfigurationId]
+                );
+
+                return new NotFoundResponse('Agent Configuration');
+            }
```

Comment:
One common practice in repositories is to have: 
- `find` method when you can return object|null
- `get` when you expect an object but not null

When you call `get` and have no result, your repository can throw an exception (catchable in the use case for the moment or later on) and handle the error log around the repository and not in each use case calling the repo method.